Criteo audience doc (#1143)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/criteo-audience.mdx

| Connection Mode | Web       | Mobile | Server |
| :------------------ | :------------ | :--------- | :--------- |
| **Device mode**     | - | -          | -          |
| **Cloud mode**      | **Supported**   | **Supported**  |  -  |

Once you have confirmed that the source platform supports sending events to Criteo Audience, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Criteo Audience**.
2. Assign a name to the destination and click **Continue**.

## Connection settings

To successfully configure Criteo Audience as a destination, you need to configure the following settings:

- **Create Account**: Click **Create Account** > **Connect with Criteo Audience** and enter your login credentials for Criteo Audeince.
- **Audience Id**: Enter the ID of the audience for which you want to sync the data. Refer to the [Criteo Audience](https://developers.criteo.com/marketing-solutions/docs/audience#creating-a-new-audience) documentation for more information on obtaining the audience ID.

> - **Audience Type**: Enter the type of audience for which you want to add or remove the users.

Reviewer: ```suggestion
- **Audience Type**: Enter the type of audience by which you want to add or remove the users.
```

## `audienceList` event structure

RudderStack supports <a href="https://developers.criteo.com/marketing-solutions/docs/contact-list#adding-and-removing-users-in-an-audience">adding and removing users in an audience list</a>.

The following code snippet shows a sample `audienceList` call:

```json
{
  "data": {
    "type": "ContactlistAmendment",
    "attributes": {
      "operation": "add" / "remove",
      "identifierType": "email",
      "identifiers": [
        "example1@gmail.com"
      ],
      "gumCallerId": "",
      "internalIdentifier": "false"
    }
  }
}
```

In the above snippet, you can pass the `identifierType` as:
- `email`: Email Address
- `madid`: Mobile Ad Identifier
- `identityLink`: LiveRamp Identity Link
- `gum`: Criteo GUM cookie identifier

> 

Reviewer: Can we please add a note somewhere that users must add the following redirect URLn their Application Details:
https://app.rudderstack.com/auth/destination/criteo_audience/callback

Author: added

---
title: "Criteo Audience"
description: Send your event data from RudderStack to Criteo Audience.
---

> [Criteo](https://www.criteo.com/) is an online display advertising platform and its [Audience API](https://developers.criteo.com/marketing-solutions/docs/audiences) lets you manage your contact lists by creating, deleting, and updating audience data, which is used for your advertising campaigns.

Reviewer 2: ```suggestion
[Criteo](https://www.criteo.com/) is an online display advertising platform. Its [Audience API](https://developers.criteo.com/marketing-solutions/docs/audiences) lets you manage your contact lists by creating, deleting, and updating audience data, which you can use for your advertising campaigns.
```

Find the open source transformer code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/src/v0/destinations/Criteo Audience_audience">GitHub repository</a>.

## Getting started

RudderStack supports sending event data to Criteo Audience via the following <a href="https://rudderstack.com/docs/rudderstack-cloud/rudderstack-connection-modes/">connection modes</a>:

| Connection Mode | Web       | Mobile | Server |
| :------------------ | :------------ | :--------- | :--------- |
| **Device mode**     | - | -          | -          |
| **Cloud mode**      |  -  | -  |  **Supported**  |

Once you have confirmed that the source platform supports sending events to Criteo Audience, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Criteo Audience**.
2. Assign a name to the destination and click **Continue**.

## Connection settings

To successfully configure Criteo Audience as a destination, you need to configure the following settings:

> - **Create Account**: Click **Create Account** > **Connect with Criteo Audience** and enter your login credentials for Criteo Audeince.

Reviewer 2: ```suggestion
- **Create Account**: Click **Create Account** > **Connect with Criteo Audience** and enter your login credentials for Criteo Audience.
```

> - **Audience Id**: Enter the ID of the audience for which you want to sync the data. Refer to the [Criteo Audience](https://developers.criteo.com/marketing-solutions/docs/audience#creating-a-new-audience) documentation for more information on obtaining the audience ID.

Reviewer 2: ```suggestion
- **Audience Id**: Enter the audience ID for which you want to sync the data. Refer to the [Criteo Audience documentation](https://developers.criteo.com/marketing-solutions/docs/audience#creating-a-new-audience) for more information on obtaining the audience ID.
```

> - **Audience Type**: Select the type of audience from the dropdown for which you want to add or remove the users. If you select `gum`, then **Gum Caller Id** is a required field. Refer to the [Criteo documentation](https://developers.criteo.com/marketing-solutions/docs/contact-list#adding-and-removing-users-in-an-audience) for more information on different audience types.

Reviewer 2: ```suggestion
- **Audience Type**: Select the audience type from the dropdown for which you want to add or remove the users. If you select `gum`, then **Gum Caller Id** is a required field. Refer to the [Criteo documentation](https://developers.criteo.com/marketing-solutions/docs/contact-list#adding-and-removing-users-in-an-audience) for more information on different audience types.
```

## `audienceList` event structure

RudderStack supports <a href="https://developers.criteo.com/marketing-solutions/docs/contact-list#adding-and-removing-users-in-an-audience">adding and removing users in an audience list</a>.

The following code snippet shows a sample `audienceList` call for `email` audience type:

```json
{
  "type": "audiencelist",
  "properties": {
    "listData": {
      "add": [{
          "email": 'alex@example.com'
        },
        {
          "email": 'john@example.com'
        }
      ]
    }
  }
}
```

The following code snippet shows a sample `audienceList` call for `madid` audience type:

```json
{
  "type": "audiencelist",
  "properties": {
    "listData": {
      "add": [{
          "madid": "sample_madid"
        },
        {
          "madid": "sample_madid_1"
        }
      ]
    }
  }
}
```

> Criteo's API implements [rate limits](https://developers.criteo.com/marketing-solutions/docs/rate-limits) to limit the number of requests you're able to perform on any API endpoint per minute.

Reviewer 2: ```suggestion
<div class="warningBlock">
Criteo's API implements [rate limits](https://developers.criteo.com/marketing-solutions/docs/rate-limits) to limit the number of requests you're able to perform on any API endpoint per minute.
</div>

```
